dataset/src/test/scala/frameless/functions/UdfTest.scala

```diff
@@ -0,0 +1,251 @@
+package frameless.functions
+
+import frameless.{TypedDataset, TypedDatasetSuite, TypedEncoder, X3, X5}
+import org.scalacheck.Prop
+import org.scalacheck.Prop._
+
+import scala.reflect.ClassTag
+import scala.reflect.runtime.universe._
+import frameless.functions.udfs._
+
+class UdfTest extends TypedDatasetSuite {
+
+  test("one argument udf") {
+    def prop[A, B, C](f1: A => A)(data: Vector[X3[A, B, C]])(
+      implicit
+      ea: TypedEncoder[A],
+      ex4: TypedEncoder[X3[A, B, C]],
+      ta: TypeTag[A],
+      ca: ClassTag[A]
+    ): Prop = {
+      val dataset = TypedDataset.create(data)
+      val u1 = udf[X3[A, B, C], A, A](f1)
+      val u2 = dataset.makeUDF(f1)
+      val A = dataset.col[A]('a)
+
+      val dataset21 = dataset.select(u1(A)).collect().run().toVector
+      val dataset22 = dataset.select(u2(A)).collect().run().toVector
+      val d = data.map(x => f1(x.a))
+
+      dataset21 ?= d
+      dataset22 ?= d
```

Comment:
Here we are testing the two ways you can create a udf using our udf function and the makeUDF method from TypedDataset.